plenum/server/replica.py

```diff
@@ -552,14 +553,18 @@ def _setup_for_non_master(self):
                 else lowest_prepared - 1
             self.logger.debug('{} Setting last ordered for non-master as {}'.format(
                 self, self.last_ordered_3pc))
-            self.last_ordered_3pc = (self.viewNo, lowest_ordered)
-            self._clear_last_view_message_for_non_master(self.viewNo)
+            # now after catch up we have in last_ordered_3pc value from
+            # master replica and should change last_ordered_3pc
+            # to lowest_ordered or 0
+            if is_catchup or lowest_ordered > self.last_ordered_3pc[1]:
+                self.last_ordered_3pc = (self.viewNo, lowest_ordered)
+                self.update_watermark_from_3pc()
```

Comment:
We must not update the watermarks in case we have not found any prepared certificate. Otherwise we actually reset the watermarks to `(0, LOG_SIZE)`. This may lead to that all incoming 3PC-messages will be stashed as laying outside of watermarks and so we will never gather a prepared certificate.

Now in case the flag `is_catchup` is `True` we update the watermarks even if we have not found any prepared certificate.